core/templates/dev/head/pages/exploration_player/ProgressNavDirective.js

```diff
@@ -47,13 +68,12 @@ oppia.directive('progressNav', [
             }
           };
 
-          $scope.hasPrevious = function() {
-            return $scope.currentCardIndex > 0;
-          };
-
-          $scope.hasNext = function() {
-            return $scope.currentCardIndex < $scope.transcriptLength - 1;
-          };
+          $scope.showContinueButton = function() {
+            var lastPair = $scope.activeCard.inputResponsePairs[
+              $scope.activeCard.inputResponsePairs.length - 1];
+            return $scope.isInteractionInline &&
+              $scope.activeCard.destStateName && lastPair.oppiaResponse;
```

Comment:
Should this be lastPair.oppiaResponse !== null, or are you including an empty string as well?

Also, suggest casting to Boolean explicitly: `return Boolean(...);`. Otherwise I think it takes on the last value.

Also, just to check, is this a minimal criterion? I.e. are all three clauses necessary, and removing any one of them makes this check wrong?

Reply:
Looks like it should be an undefined check, fixed. Though interestingly [this code](https://github.com/oppia/oppia/blob/4c0a9dfa559c7a183eb26cb12aec437ab6122ac8/core/templates/dev/head/pages/exploration_player/tutor_card_directive.html#L71) doesnt check against undefined?

Yes, all three should be true to show the continue button.

Reply:
Ah, I didn't realize there was prior art. Let's keep it as it was then (no need for introducing !== undefined here). In the future we should get oppiaResponse to be a domain object and then this problem will go away. Sorry for the churn.